meta-rz-bsp/recipes-bsp/u-boot/files/0002-configs-smarc-rzg2l-g2lc-fix-SD-card-boot-and-set-it.patch

```diff
@@ -0,0 +1,91 @@
+From 19fead31c83e15f32a1ee68997d00dd6be465bcd Mon Sep 17 00:00:00 2001
+From: Hugo Villeneuve <hvilleneuve@dimonoff.com>
+Date: Mon, 15 Apr 2024 11:24:24 -0400
+Subject: [PATCH 1/2] configs: smarc-rzg2l/g2lc: fix SD card boot and set it as
+ default
+
+Make SD card boot by default.
+
+Support SD card image with a single partition (rootfs, no boot partition).
+---
+ include/configs/smarc-rzg2l.h  | 19 +++++++++++++------
+ include/configs/smarc-rzg2lc.h | 19 +++++++++++++------
+ 2 files changed, 26 insertions(+), 12 deletions(-)
+
+diff --git a/include/configs/smarc-rzg2l.h b/include/configs/smarc-rzg2l.h
+index b6d2c8af33..d91a9cb55d 100644
+--- a/include/configs/smarc-rzg2l.h
++++ b/include/configs/smarc-rzg2l.h
+@@ -57,19 +57,26 @@
+ /* The HF/QSPI layout permits up to 1 MiB large bootloader blob */
+ #define CONFIG_BOARD_SIZE_LIMIT		1048576
+ 
++#define MMC_DEV		__stringify(0)	/* SD card on SOM */
++#define ROOT_DEV	"mmcblk0"
++#define ROOT_PART	__stringify(1)	/*
++					 * rootfs on partition 1. If using a
++					 * separate boot partition, set to 2.
++					 */
++
+ /* ENV setting */
+ 
+ #define CONFIG_EXTRA_ENV_SETTINGS \
++	"mmc_dev=" MMC_DEV "\0" \
++	"root_dev=" ROOT_DEV "\0" \
++	"root_part=" ROOT_PART "\0" \
+ 	"usb_pgood_delay=2000\0" \
+ 	"bootm_size=0x10000000 \0" \
+-	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
+-	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
++	"prodbootargs=setenv bootargs rw rootwait earlycon root=/dev/${root_dev}p${root_part}\0" \
+ 	"bootimage=unzip 0x4A080000 0x48080000; booti 0x48080000 - 0x48000000 \0" \
+-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a07g044l2-smarc.dtb;run prodemmcbootargs \0" \
+-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a07g044l2-smarc.dtb;run prodsdbootargs \0" \
+-	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
++	"mmcload=ext4load mmc ${mmc_dev}:${root_part} 0x48080000 boot/Image;ext4load mmc ${mmc_dev}:${root_part} 0x48000000 boot/r9a07g044c2-smarc.dtb; run prodbootargs\0"
```

Comment:
This should be `r9a07g044l2-smarc.dtb` for G2L.

Reply:
Ok, will fix it.